File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/nomenclator/CustomNomenclatorSelectionFiltersResponseModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.nomenclator;

import java.util.List;
import java.util.Map;

public class CustomNomenclatorSelectionFiltersResponseModel {
	
	private boolean selectable;
	private List<Long> valueIds;
	private Map<String, List<String>> attributeValuesByKey;
	
	public boolean isSelectable() {
		return selectable;
	}
	
	public void setSelectable(boolean selectable) {
		this.selectable = selectable;
	}
	
	public List<Long> getValueIds() {
		return valueIds;
	}
	
	public void setValueIds(List<Long> valueIds) {
		this.valueIds = valueIds;
	}
	
	public Map<String, List<String>> getAttributeValuesByKey() {
		return attributeValuesByKey;
	}
	
	public void setAttributeValuesByKey(Map<String, List<String>> attributeValuesByKey) {
		this.attributeValuesByKey = attributeValuesByKey;
	}
}

File: CloudFlow/FrontEnd-App/src/app/shared/model/nomenclator/custom-nomenclator-selection-filters-response.model.ts
import { JsonObject, JsonProperty } from "json2typescript";

@JsonObject
export class CustomNomenclatorSelectionFiltersResponseModel {

	@JsonProperty("selectable", Boolean)
	public selectable: boolean = null;

	@JsonProperty("valueIds", [Number])
	public valueIds: number[] = null;

	@JsonProperty("attributeValuesByKey", Object)
	public attributeValuesByKey: object = null;
}